### PR TITLE
doxygen: fix various build warnings

### PR DESCRIPTION
**Pull request by JordanYates**
Fix the following doc build warning on main:
```
WARNING: /home/jordan/code/zephyr/zephyr/include/zephyr/drivers/firmware/scmi/util.h:227: The following parameters of DT_INST_SCMI_PROTOCOL_DEFINE(inst, init_fn, pm, data, config, level, prio, api) are not documented:
WARNING: /home/jordan/code/zephyr/zephyr/include/zephyr/devicetree.h:4532: The following parameter of DT_ANY_COMPAT_HAS_PROP_STATUS_OKAY(compat, prop) is not documented:
WARNING: /home/jordan/code/zephyr/zephyr/include/zephyr/llext/llext.h:296: unexpected token in comment block while parsing the argument of command retval
```

Another instance of #72605

### CODE REVIEW

File: include/zephyr/drivers/firmware/scmi/util.h [hunk 1]
```diff
@@ -221,7 +221,13 @@
...
- * @param ... other parameters as expected by DT_SCMI_PROTOCOL_DEFINE()
+ * @param init_fn pointer to protocol's initialization function
+ * @param api pointer to protocol's subsystem API
+ * @param pm pointer to the protocol's power management resources
+ * @param data pointer to protocol's private data
+ * @param config pointer to protocol's private constant data
+ * @param level protocol initialization level
+ * @param prio protocol's priority within its initialization level
  */
 #define DT_INST_SCMI_PROTOCOL_DEFINE(inst, init_fn, pm, data, config,		\
```
Comment (gmarull) [inline include/zephyr/drivers/firmware/scmi/util.h:232]: ugh, DT namespace used outside of `devicetree.h`

**Review by gmarull — DISMISSED** [review]

**Review by pillo79 — COMMENTED** [review]
Thanks for fixing the `@kconfig{}` warning.
Having both `@returns` and `@retval` is a very common pattern in `include/zephyr/llext/` though; that way we have a generic case, but LLEXT-specific errors are documented in more detail. 
If this is to be avoided, should all other uses be amended similarly?

**Review by decsny — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]
llext stuff looks fine

**Comment by JordanYates** [comment]
> Thanks for fixing the `@kconfig{}` warning. Having both `@returns` and `@retval` is a very common pattern in `include/zephyr/llext/` though; that way we have a generic case, but LLEXT-specific errors are documented in more detail. If this is to be avoided, should all other uses be amended similarly?

My understanding from previous feedback is that code should be using one or the other, not both. I can't find any published information on this policy though. I think it was from @gmarull?

**Comment by kartben** [comment]
> > Thanks for fixing the `@kconfig{}` warning. Having both `@returns` and `@retval` is a very common pattern in `include/zephyr/llext/` though; that way we have a generic case, but LLEXT-specific errors are documented in more detail. If this is to be avoided, should all other uses be amended similarly?
> 
> My understanding from previous feedback is that code should be using one or the other, not both. I can't find any published information on this policy though. I think it was from @gmarull?

I've started to assemble some long overdue Doxygen guidelines and plan to basically recommend something along the lines of what can be found in e.g. RTEMS documentation: https://docs.rtems.org/branches/master/eng/coding-doxygen.html#function-declarations

> Distinctive return values shall be documented with an `@retval` entry. Document the most common return value first. Use `@return` to describe the return of non-distinctive values. Use grammatically correct sentences for the descriptions. Use sentences in simple past tense to describe conditions which resulted in the return of a status value. Place `@retval` descriptions before the `@return` description. For functions returning a boolean value, use `@return` and a phrase like this: “Returns true, if some condition is satisfied, otherwise false.”.

```
/**
 * @brief Allocates a memory block of the specified size from the workspace.
 *
 * @param size is the size in bytes of the memory block.
 *
 * @retval NULL No memory block with the requested size was available in the
 *   workspace.
 *
 * @return Returns the pointer to the allocated memory block, if enough
 *   memory to satisfy the allocation request was available.  The pointer is at
 *   least aligned by #CPU_HEAP_ALIGNMENT.
 */
void *_Workspace_Allocate( size_t size );
```

**Review by pillo79 — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Pull request merged**